github4s/src/main/scala/github4s/domain/Repository.scala

```diff
@@ -371,17 +371,83 @@ object RepoUrlKeys {
       files: Seq[FileComparison] = Seq.empty
   )
 
-  final case class FileComparison(
-      sha: String,
-      filename: String,
-      status: String,
-      additions: Int,
-      deletions: Int,
-      changes: Int,
-      blob_url: String,
-      raw_url: String,
-      contents_url: String,
-      patch: String
-  )
+  /**
+   * A file comparison that contains information on the changes to a file.
+   * There are two subtypes: `FileComparisonNotRenamed` and `FileComparisonRenamed` that have different guarantees.
+   *
+   * * `FileComparisonNotRenamed` guarantees that the `patch` field exists, does not have a `previous_filename` field.
+   * * `FileComparisonRenamed` guarantees that the `previous_filename` field exists and sometimes contains a `patch` field.
+   *
+   * To get values from these fields, there are helper methods `getPatch` and `getPreviousFilename`, though
+   * it is recomended to match on your `FileComparison` value to determine which type it is, to remove ambiguity.
+   */
+  sealed trait FileComparison {
+    def sha: String
+    def filename: String
+    def status: String
+    def additions: Int
+    def deletions: Int
+    def changes: Int
+    def blob_url: String
+    def raw_url: String
+    def contents_url: String
+
+    /**
+     * Gets the contents of the `patch` field if it exists, in the case that the file was modified.
+     * To guarantee that the `patch` field is available, match this `FileComparison` value as a
+     * `FileComparison.FileComparisonNotRenamed` type which always has this field.
+     */
+    def getPatch: Option[String]
```

Comment:
Would it make sense to remove the `getPatch` and `getPreviousFilename` fields from here? If the user wants these fields, they must match the concrete type. If they don't need those fields, they can use the supertype.

Reply:
I think it's a lot less verbose to use these rather than do a proper pattern match for some cases. If I'm using this library and I want "the previous filename if it exists" then that's a single method call. Or if I want the patch "if it exists" that's also a single call. Takes up a lot less vertical space. The main use case of doing the pattern match would be if you're passing the file comparison to some downstream function and want to validate that it's the right subtype first, but I don't think it should be required if we can help it.